### PR TITLE
Tooltip fixes

### PR DESCRIPTION
**Pull request by DxCx**
Hey @madou,
I'm helping @DomGrieco with tooltips,
here are a bunch of fixes :)

### CODE REVIEW

**Comment by DomGrieco** [comment]
Looks good!

**Comment by madou** [comment]
howd you find setting up the project? easy? anything you'd suggest updating in the readme to make some things more obvious?

File: scripts/start.js [hunk 1]
```diff
@@ -74,11 +77,12 @@ function setupCompiler (port) {
     clearConsole();
     const hasErrors = stats.hasErrors();
     const hasWarnings = stats.hasWarnings();
+
     if (!hasErrors && !hasWarnings) {
       console.log(chalk.green('Compiled successfully!'));
       console.log();
       // eslint-disable-next-line
-      console.log('The app is running at http://localhost:' + port + '/');
+      console.log('The app is running at http://localhost:' + port + basePath);
```
Comment (madou) [inline scripts/start.js]: awesome 💯

File: src/common/components/Tooltip/Fact/index.js [hunk 2]
```diff
@@ -28,13 +36,25 @@ type FactProps = {
...
+    dmg_multiplier: number,
     prefix: {
       status: string,
       icon: string,
     },
   },
 };
 
+const ATTRIBUTE_MAPPING = {
+  ConditionDamage: 'Condition Damage',
```
Comment (madou) [inline src/common/components/Tooltip/Fact/index.js]: this all the mapping we need? can you also apply translations here please :)

```
import T from 'i18n-react';

T.translate('itemAttributes.conditionDamage');
```

etc

File: src/common/components/Tooltip/Item/index.js [hunk 3]
```diff
@@ -31,6 +31,16 @@ function buildName (item, skin, upgrades) {
   return prefixedName;
 }
 
+const ATTRIBUTE_MAPPING = {
```
Comment (madou) [inline src/common/components/Tooltip/Item/index.js]: oh, more mappings :-)

same comment as above, though (translations please !)

File: src/common/components/Tooltip/Item/index.js [hunk 4]
```diff
@@ -31,6 +31,16 @@ function buildName (item, skin, upgrades) {
...
+  BoonDuration: 'Concentration',
+  ConditionDuration: 'Expertise',
+  CritDamage: 'Ferocity',
+};
+
+function attributeName (attribute) {
+  return ATTRIBUTE_MAPPING[attribute] || startCase(attribute);
```
Comment (madou) [inline src/common/components/Tooltip/Item/index.js]: 👏

**Comment by madou** [comment]
cool thats all my comments for now! are you planning on finishing all the tooltip fixes in this PR, or you'll make another one for the rest?

**Comment by DxCx** [comment]
will take a look at all of your comments later. thanks for the review @madou

**Comment by madou** [comment]
no worries @DxCx , let me know if you need anything

**Comment by DomGrieco** [comment]
Website is planned to release sometime on the 5th (evening most likely). It would be sweet if we could get this pushed by then

**Comment by DxCx** [comment]
Ok, ill try to work on the fixes later today.. :)
Didnt realized its urgent :)

**Comment by DomGrieco** [comment]
It's not urgent but it would be sweet if when we released the site these changes were live

File: scripts/start.js (hunk 1 above)
Comment (DxCx) [inline scripts/start.js]: well, needing to dig for the url for the embeded html was the most annoying part,
that's why i had to add this.. (regarding how was the project set up)

File: src/common/components/Icon/index.js [hunk 5]
```diff
@@ -12,10 +12,17 @@ type IconProps = {
...
+  applyCount?: number,
 };
 
-const Icon = ({ name, size, className, src, button, children, ...props }: IconProps) => {
+const Icon = ({ name, size, className, src, button, children, applyCount, ...props }: IconProps) => {
+  const APPLY_COUNT_THRESHOLD = 1;
   let image: string;
+  let applyBadge: any;
+
+  if (applyCount && applyCount > APPLY_COUNT_THRESHOLD) {
```
Comment (DxCx) [inline src/common/components/Icon/index.js]: Initially that was something like this,
and then i taught we might want to add an apply count for any icon, so why not adding it globally.

please checkout the branch and see how it looks like, but i really think it should just be icon property to make sure any need in the future will keep the same logic & css.

File: src/common/components/Tooltip/Fact/index.js (hunk 2 above)
Comment (DxCx) [inline src/common/components/Tooltip/Fact/index.js]: i tought about running throu translation and then i realized that i don't know all possablities for those mappings..
where can i find all possablities so i can make sure all the values matches with existing translation names?

**Comment by DxCx** [comment]
@madou 
translations are fixed, regarding the number on top of icon, i really think it should be icon property.
maybe the naming is wrong as it might be used for more then apply count, but this is essentially the number on top of icon like in game.

regarding the other fixes,
i think it's for the best to merge this so @DomGrieco can continue his amazing QA work and test it.
so we will open another PR for the others.

File: scripts/start.js (hunk 1 above)
Comment (madou) [inline scripts/start.js]: yeah thats my bad, i didnt really get time to set up the dev experience when i made this (was a very recent change :))

File: src/common/components/Icon/index.js (hunk 5 above)
Comment (madou) [inline src/common/components/Icon/index.js]: agreed - icon is used for more than just gw2 icons though, its used for any images really.

maybe we can make a Gw2Icon that wraps Icon? at least it'd be more focused. im happy to ignore it for now and do it later though.

File: src/common/components/Tooltip/Fact/index.js [hunk 6]
```diff
@@ -75,10 +113,11 @@ const Fact = ({ data }: FactProps) => {
       break;
 
     case 'BuffConversion':
+      // XXX: String itself should be translated.
```
Comment (madou) [inline src/common/components/Tooltip/Fact/index.js]: there isn't one atm, but if you add the translation into every text file (under `lib/i18n/texts`), should be fine. we dont need any translations atm really, but just at least something in the texts so someone can replace them at a later date !

**Comment by madou** [comment]
if you want im happy to merge this and fix things up later so dom gets his changes faster. just fix the merge conflict and we're good to go

**Comment by DxCx** [comment]
@madou  - Done. i belive it can be merged.

**Pull request merged**